### PR TITLE
Multiple cleanup items for CI LAB

### LINKED ISSUE #23: Update to use OSAL socket API
**Is your feature request related to a problem? Please describe.**
Currently uses cFE supplied network_includes.h and not the abstracted OSAL calls

**Describe the solution you'd like**
Update to use OSAL

**Describe alternatives you've considered**
None

**Additional context**
See osapi-os-net.h and/or OSAL API.

**Requester Info**
Jacob Hageman - NASA/GSFC

### LINKED ISSUE #27: Remove CFDP PDU introspection features
**Is your feature request related to a problem? Please describe.**
CFDP PDU introspection is old debug code.  Remove to simplify ci_lab.

Also subject of recommended issues from lgtm:

ci_lab_app.c
```
line 625 - 732:
Switch has at least one case that is too long: 4 (37 lines)

line 599:

/*    OS_printf("CI: Captured PDU with length = %d\n",PduHdrPtr->PDataLen); */
--
  | This comment appears to contain commented-out code

```

**Requester Info**
Anh Van, NASA Goddard

### LINKED ISSUE #34: CI_LAB should split message handlers into separate functions
**Is your feature request related to a problem? Please describe.**
The "process ground command" routine in CI_LAB processes some commands "inline".  This does not match the current/recommended style which is to put command handling into a separate function.

**Describe the solution you'd like**
Putting each command into a separate function is the preferred CCB style and it improves modularity of the code.

**Additional context**
All CFE core apps the sample_app all implement the preferred pattern.  CI_LAB should be consistent.

**Requester Info**
Joseph Hickey, Vantage Systems, Inc.

### PR DESCRIPTION
**Pull request by jphickey**
**Describe the contribution**

Cleans up multiple aspects of the CI_LAB implementation.  Details Below.

Fixes #23
Fixes #27
Fixes #34

These change sets are submitted as a single pull request, as they have inter-dependencies between them and will trigger conflicts if presented individually.  This pull request still contains each fix as an individual commit, so each one can be independently reviewed.

For #27 / commit  157aac0: PDU introspection code in CI LAB has numerous quality issues and has repeatedly been a maintenance problem, and also is of questionable usefulness.  It was agreed in the CCB to drop this feature.

For #23 / commit 3ff5cc1: Use the OSAL Socket API instead of calling BSD sockets functions directly.  This is fairly straightforward.

For #34 / commit 5f873e0: Update the CI_LAB local command/telemetry processing logic to match the recommended patterns used in other modules.  This applies the recommended naming conventions, message structure, and functional structure by giving each command a dedicated handler function.

**Testing performed**
Build for native for both debug/release builds, with unit tests enabled.  Confirmed normal operation of CI_LAB for command ingest by running CFE and sending commands with cmdUtil.

**Expected behavior changes**
CFDP PDU manipulation with CI LAB is no longer supported. Normal command ingest behavior does not change.

**System(s) tested on**
Ubuntu 18.04 LTS 64 bit

**Contributor Info - All information REQUIRED for consideration of pull request**
Joseph Hickey, Vantage Systems, Inc.

### CODE REVIEW

**Comment by jphickey** [comment]
Re-pushed branch to address some minor remaining cleanup items and also add a fix for #36 to the mix.  Like the others, this needs to be sequential; all changes in this pull request depend on the one before it, and cannot be cherry-picked/rebased without conflicts.  But they can be independently reviewed.

**Comment by jphickey** [comment]
Updated the changeset to put the event generator in the actual Reset command handler as opposed to the internal reset routine.  This avoids a spurious reset event reported at startup.

File: fsw/src/ci_lab_app.c [hunk 1]
```diff
@@ -203,582 +217,180 @@ void CI_TaskInit(void)
...
+    CFE_SB_TimeStampMsg(&CI_LAB_Global.HkBuffer.MsgHdr);
+    CFE_SB_SendMsg(&CI_LAB_Global.HkBuffer.MsgHdr);
+    return CFE_SUCCESS;
 
-} /* End of CI_ReportHousekeeping() */
+} /* End of CI_LAB_ReportHousekeeping() */
 
 /* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * **/
-/*  Name:  CI_ResetCounters                                                   */
+/*  Name:  CI_LAB_ResetCounters                                               */
```
Comment (skliper) [inline fsw/src/ci_lab_app.c]: CI_LAB_ResetCounters_Internal

Comment (astrogeco) [reply fsw/src/ci_lab_app.c]: @skliper are you requesting this name be changed?

Comment (skliper) [reply fsw/src/ci_lab_app.c]: correct.

**Comment by jphickey** [comment]
See commit 07a7077, this updates the comment for `CI_LAB_ResetCounters_Internal`.  Did not see any `CI_ResetCounters` in the final commit of the pull request so possible that one was already changed.

**Pull request merged**